tt_metal/impl/buffers/buffer.hpp

```diff
@@ -207,6 +208,8 @@ class Buffer {
 
     bool is_l1() const { return buffer_type() == BufferType::L1 or buffer_type() == BufferType::L1_SMALL; }
     bool is_dram() const { return buffer_type() == BufferType::DRAM; }
+    bool is_trace() const { return buffer_type() == BufferType::TRACE; }
+    bool allocated_in_dram() const { return this->is_trace() or this->is_dram(); }
```

Comment:
Should we keep l1/dram functions consistent?

is_l1 returns true for l1 or l1 small, should is_dram behave as how you have allocated_in_dram instead? Or vice versa?

Reply:
I tried to add a separation between trace and DRAM buffers, in case we ever decide to allocate trace in L1 (highly unlikely). I guess for consistency your suggestion does make sense. I'll make the requested changes.